subxt/src/storage/storage_key.rs

```diff
@@ -12,19 +12,18 @@ use derivative::Derivative;
 
 use super::utils::hash_bytes;
 
-#[derive(Debug, Clone)]
-// An iterator over all type ids of the key and the respective hashers.
-pub struct StorageHashersIter {
+/// A collection of storage hashers paired with the type ids of the types they should hash.
+/// Can be created for each storage entry in the metadata via [`StorageHashers::new()`].
+#[derive(Debug)]
+pub(crate) struct StorageHashers {
     hashers_and_ty_ids: Vec<(StorageHasher, u32)>,
-    idx: usize,
 }
 
-impl StorageHashersIter {
-    /// Creates a new [`StorageHashersIter`]. Looks at the [`StorageEntryType`] and
+impl StorageHashers {
+    /// Creates new [`StorageHashers`] from a storage entry. Looks at the [`StorageEntryType`] and
     /// assigns a hasher to each type id that makes up the key.
     pub fn new(storage_entry: &StorageEntryType, types: &PortableRegistry) -> Result<Self, Error> {
```

Comment:
pub(crate) perhaps so we don't need to care about this being in the public API? :)

Reply:
Ah I see; the whole thing is now!